app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java

```diff
@@ -939,7 +939,11 @@ public static boolean isDataStale() {
     }
 
     public static BgReading readingNearTimeStamp(double startTime) {
-        final double margin = (4 * 60 * 1000);
+        double margin = (4 * 60 * 1000);
+        return readingNearTimeStamp(startTime, margin);
+    }
+
+    public static BgReading readingNearTimeStamp(double startTime, final double margin) {
```

Comment:
FYI there is: `getForPreciseTimestamp` but it has slightly different search criteria so I'm not certain if it is what you need here. `readingNearTimeStamp` should probably at some point also be refactored to use `long` instead of `double`